### ISSUE TITLE
KT-30596 - Kotlin compiler warnings don't show up in build scans

### ISSUE THREAD

**Issue by wolfs** [gradle/kotlin-dsl-samples#1031]
https://youtrack.jetbrains.com/issue/KT-30596

When using Kotlin in `buildSrc` compiler warnings should show up in build scans.

### Expected Behavior
The warning I see on the command line
```
w: /Users/wolfs/projects/gradle/build-tool-release/buildSrc/subprojects/configuration/src/main/kotlin/org/gradle/gradlebuild/dependencies/DependenciesMetadataRulesPlugin.kt: (75, 54): Unchecked cast: Any? to List<CapabilitySpec>
```
should show up in the corresponding build scan, too: https://e.grdev.net/s/h3q7axo67mcya/console-log

### Current Behavior
No Kotlin compiler warnings show up in the build scan.

### Steps to Reproduce (for bugs)
Run `./gradlew compileAll` in the `gradle/gradle` build.

### Your Environment
 * Build scan URL: https://e.grdev.net/s/h3q7axo67mcya

**Comment by eskatos** [comment]
These warnings are emitted by the `kotlin-gradle-plugin`

**Comment by ldaley** [comment]
My guess is that they are coming from a forked thread, with no bound build operation.

**Comment by eskatos** [comment]
The  `kotlin-gradle-plugin` runs the Kotlin compiler either in-process or in a forked kotlin compiler daemons.

By default it runs in a Kotlin compiler daemon, with no bound build operation.
That's why the warnings don't show up in build scans.

When running "in-process", then the warnings do show up in build scans:
https://scans.gradle.com/s/275h5cmj7b42o/console-log#L3

**Comment by eskatos** [comment]
Fixed by https://github.com/JetBrains/kotlin/pull/2211
To be released in 1.3.40
